### PR TITLE
Unify the Cudnn Execution Plan Builder

### PR DESCRIPTION
**Pull request by kaixih**
Note: This PR is on the top of https://github.com/tensorflow/tensorflow/pull/53843.

This PR tries to unify the code paths from the cudnn op graph to the execution plans for both unfused and fused conv ops. Both code paths follow: build op graph -> query heuristics/fallback plans -> filter out undesired plans (e.g. non-deterministic or winograd engines) -> filter out errata plans (e.g. engines that might be buggy).

cc. @nluehr

### CODE REVIEW

File: tensorflow/stream_executor/cuda/cuda_dnn.cc [hunk 1]
```diff
@@ -4565,6 +4565,117 @@ class ScalingParam {
   float as_float_;
 };
 
+namespace {
```
Comment (awpr) [inline tensorflow/stream_executor/cuda/cuda_dnn.cc:4568]: This might need a CUDNN_VERSION / TF_ENABLE_CUDNN_FRONTEND preprocessor guard?

**Review by awpr — COMMENTED** [review]
Overall looks good, glad to see the duplication steadily decreasing.  Just a few minor nits.

**Comment by kaixih** [comment]
Thanks for the comments. I have committed the new change (the top one). PTAL. @awpr

**Review by awpr — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by kaixih** [comment]
Same with this one. I see "Windows Bazel GPU" fails but the details are not accessible. @awpr @jurahul

**Review by awpr — APPROVED** [review]

**Review by awpr — APPROVED** [review]

**Comment by kaixih** [comment]
@awpr I noticed the test below is failed. Can you help with it?
```
import/copybara — An error happened while migrating the change
```

**Comment by awpr** [comment]
Still trying to figure out how to get it to retry.

**Pull request merged**